src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -252,7 +252,8 @@ protected CliGitAPIImpl(String gitExe, File workspace, TaskListener listener, En
         this.listener = listener;
         this.gitExe = gitExe;
         this.environment = environment;
-        this.encoding = isZos() ? "IBM1047" : Charset.defaultCharset().toString();
+        this.encoding = isZos() && System.getProperty("ibm.system.encoding")!=null ? 
```

Comment:
I don't think that does what is needed in the rest of the file.

The rest of the file acts as though this.encoding is the String representation of a Charset name, yet this assignment will cause `this.encoding` to be assigned the value `true` on zOS and `false` on all other platforms (at least as far as I can tell).

I assume that methods which are using `encoding` continue to pass the unit tests because the called methods default to the system encoding when they detect an unrecognized encoding character set.

Please retain the prior meaning of this.encoding, since it is used in other places in the file.

Reply:
Mmm... tertiary operations should take precedence if i'm not mistaken
<img width="1272" alt="image" src="https://user-images.githubusercontent.com/1420875/46089291-d9f94200-c1b6-11e8-9a8c-ee80aeea6522.png">
<img width="1279" alt="image" src="https://user-images.githubusercontent.com/1420875/46089351-f4cbb680-c1b6-11e8-9f12-ff7b1a2effd7.png">

Reply:
but agree, better to rewrite for readability